MapShed BMPs: Reclassify Developed Land Categories, and other adjustments

rajadain opened the pull request:
## Overview

This reassigns NLCD 21 Open Space to go to Low Density Residential instead of Low Density Mixed, and updates the impervious percentage values for all urban land use types. Also upgrades GWLF-E, which in https://github.com/WikiWatershed/gwlf-e/pull/84 renames Low Density Residential to Low Density Open Space and adds it to the output. It also removes Other Upland Areas from the output.

Connects #2937 

### Demo

![image](https://user-images.githubusercontent.com/1430060/44693017-9c788c00-aa33-11e8-8d5a-0df13fa84d5c.png)

### Notes

Once this is approved, I will merge the GWLF-E PR and make a new release, and update the final commit here to point to that new release instead of the PR branch.

It was easier to remove "Other Upland Areas" from gwlf-e than to remove it by hand in all the various places in MMW (model results, compare view, CSV exports). This does change the output of gwlf-e, and we'll have to bump up the version number accordingly.

This new value won't show up in Subbasin because SRAT doesn't have a key for this. If they ever add support, we'll have to add that mapping here:

https://github.com/WikiWatershed/model-my-watershed/blob/5260b2e1fd7ce6dbc77e10dc4834b246e03513ff/src/mmw/mmw/settings/gwlfe_settings.py#L623-L638

https://github.com/WikiWatershed/model-my-watershed/blob/5260b2e1fd7ce6dbc77e10dc4834b246e03513ff/src/mmw/mmw/settings/gwlfe_settings.py#L640-L656

## Testing Instructions

* Check out this branch
* Install the latest version of `gwlf-e` in both `app` and `worker` by doing the following:

      $ vagrant ssh app -c 'sudo -H pip uninstall gwlf-e && cd /opt/app && sudo -H pip install -r requirements/development.txt && sudo restart mmw-app'
      $ vagrant ssh worker -c 'sudo -H pip uninstall gwlf-e && cd /opt/app && sudo -H pip install -r requirements/development.txt && sudo service celeryd restart'

* Go to [:8000](http://localhost:8000) and make a new MapShed project.
* Go to Water Quality results. Ensure you see Low-Density Open Space results. Ensure you don't see Other Upland Area results.
* Generate a GMS file. Ensure you see Ld_Open_Space in the file and not Ld_Residential.
* Open Compare View. Ensure the land use categories there match what you saw in model results.

kellyi commented:
Everything here is working well! Mixed-projects (e.g. those with scenarios created before and after the GWLFE changes here) also work well.

I did notice that "Other Upland Areas" is hard-coded in the GWLFE Water Quality Compare view dropdown. On `develop` that lives in `views.js` but I moved it to a `constants.js` file as part of #2940: 

https://github.com/WikiWatershed/model-my-watershed/pull/2940/commits/56936e71d067709f5d49db698b82c06d0312599b

rajadain commented:
Rebased on `develop`. Added a commit that limits the sources in Compare View to those that are on every scenario, so we won't see Low-Density Open Space in the dropdown if all scenarios don't have it. Other Upland Areas is removed completely, but it was always 0, so its removal has no ill effect.

rajadain commented:
Also added a note on how this doesn't change subbasin results because SRAT doesn't support Low-Density Open Space.

kellyi commented:
Set this up to take another look. For projects with mixed scenarios, the "Low Density Open Space" option's gone -- as expected -- from the compare view dropdown list. For entirely new projects, "Low Density Open Space" is there:

![screen shot 2018-08-28 at 6 07 14 pm](https://user-images.githubusercontent.com/4165523/44753790-51bb4a80-aaed-11e8-98b1-62188d32d047.png)

kellyi commented on src/mmw/js/src/compare/utils.js:
```diff
@@ -57,8 +58,33 @@ function mapScenariosToHydrologyTableData(scenarios) {
+                         .findWhere({ name: 'quality' })
+                         .get('result');
+
+            return _.some(result[selection.group], function(load) {
```
Totally fine to leave as is, but since `.every` is used above, this implementation could potentially also replace `_.some` and `_.filter` with the vanilla JS versions.

rajadain replied:
`scenarios` is a collection, not an array, so that `every` is actually using `underscore`'s equivalent. Thus, they are all using libraries.

rajadain commented:
Thanks for the double review!
